linalg/vector.cpp

```diff
@@ -54,6 +54,13 @@ Vector::Vector(const Vector &v)
    UseDevice(v.UseDevice());
 }
 
+Vector::Vector(Vector &&v) : data(std::move(v.data)), size(v.size)
+{
+   data.SetHostPtrOwner(v.data.OwnsHostPtr());
+   v.data.Reset();
+   v.size = 0;
+}
+
```

Comment:
Can avoid duplicating this logic
```suggestion
Vector::Vector(Vector &&v)
{
   *this = std::move(v);
}
```